joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ClassTests.scala

```diff
@@ -354,44 +354,64 @@ class ClassTests extends RubyCode2CpgFixture {
 
   }
 
-  // TODO: This should be remodelled as a property access `animal.bark = METHOD_REF`
-  "a basic singleton class" ignore {
+  "a basic singleton class" should {
     val cpg = code("""class Animal; end
         |animal = Animal.new
         |
         |class << animal
         |  def bark
         |    'Woof'
         |  end
+        |
+        |  def legs
+        |     4
+        |  end
         |end
         |
         |animal.bark # => 'Woof'
         |""".stripMargin)
 
-    "generate a type decl with the associated members" in {
-      inside(cpg.typeDecl.nameExact("<anon-class-0>").l) {
-        case anonClass :: Nil =>
-          anonClass.name shouldBe "<anon-class-0>"
-          anonClass.fullName shouldBe "Test0.rb:<global>::program.<anon-class-0>"
-          // TODO: Attempt to resolve the below with the `scope` class once we're handling constructors
-          anonClass.inheritsFromTypeFullName shouldBe Seq("animal")
-          inside(anonClass.method.l) {
-            case defaultConstructor :: bark :: Nil =>
-              defaultConstructor.name shouldBe Defines.ConstructorMethodName
-              defaultConstructor.fullName shouldBe s"Test0.rb:<global>::program.<anon-class-0>:${Defines.ConstructorMethodName}"
+    "Create assignments to method refs for methods on singleton object" in {
+      inside(cpg.method.name(":program").astChildren.isBlock.astChildren.isBlock.l) {
+        case singletonMethodsBlock :: Nil =>
+          inside(singletonMethodsBlock.astChildren.isCall.name(Operators.assignment).l) {
+            case barkAssignment :: legsAssignment :: Nil =>
+              inside(barkAssignment.argument.l) {
+                case (lhs: Call) :: (rhs: MethodRef) :: Nil =>
+                  val List(identifier, fieldIdentifier) = lhs.argument.l
```

Comment:
I speak under correction, but this might create a compilation warning, which can be suppressed as follows
```suggestion
                  val List(identifier, fieldIdentifier) = lhs.argument.l: @unchecked
```